osquery/tables/system/windows/logical_drives.cpp

```diff
@@ -19,6 +22,18 @@ QueryData genLogicalDrives(QueryContext& context) {
       "select DeviceID, Description, FreeSpace, Size, FileSystem from "
       "Win32_LogicalDisk");
   auto const& logicalDisks = wmiLogicalDiskReq.results();
+
+  const WmiRequest wmiBootConfigurationReq(
+      "select BootDirectory from Win32_BootConfiguration");
+  auto const& bootConfigurations = wmiBootConfigurationReq.results();
+  std::set<std::string> bootDirectories;
```

Comment:
In this case the order of elements in set is not important, so you can use `std::unordered_set` which is notably faster than RB tree in `std::set`.